robot_smach_states/examples/example_hear_grasp.py

```diff
@@ -9,33 +9,32 @@
 class Hear(smach.State):
     def __init__(self, robot):
         smach.State.__init__(self, outcomes=['succeeded', 'failed', 'preempted'],
-                             output_keys=['heared'])
+                             output_keys=['heard'])
         self.robot = robot
 
     def execute(self, userdata):
-        result = self.robot.ears.recognize(spec='Bring me a <drink>',
-                                           choices={'drink': ['coke', 'fanta']},
-                                           time_out=rospy.Duration(60))
-        choices = result.choices['drink'] if result else None
-        if choices:
-            rospy.loginfo('I heared:' + repr(choices))
-        userdata.heared = choices
+        answer = self.robot.hmi.query(description='can i have a fanta?',
+                                      grammar="T -> DET coke | DET fanta \n DET -> a | an ",
+                                      target='T',
+                                      timeout=60)
+        if answer:
+            rospy.loginfo('I heard:' + answer.sentence)
+        userdata.heard = answer.sentence
         return 'succeeded'
 
 
 class GetEntitysByType(smach.State):
     def __init__(self, robot):
         smach.State.__init__(self, outcomes=['succeeded', 'failed'],
-                             input_keys=['type'],
+                             input_keys=['heard'],
                              output_keys=['entity'])
         self.robot = robot
 
     def execute(self, userdata):
-        type = userdata.type
+        type = userdata.heard
 
         rospy.loginfo('I\'m going to search for {}'.format(type))
-        # call robot.ed.get_closest_entity()
-        entity = {'id': 'coke-1'}
+        entity = self.robot.ed.get_closest_entity(etype=userdata.heard)
 
         userdata.entity = entity
         return 'succeeded'
```

Comment:
As Matthijs also mentioned, this information is not used. But that was already an issue with this example. I don't think you need to fix this right now.